backend/kesaseteli/applications/tests/test_youth_applications_api.py

```diff
@@ -681,66 +693,158 @@ def test_youth_applications_detail_vtj_data_fields(
+
+    app: YouthApplication = InactiveNeedAdditionalInfoYouthApplicationFactory()
+
+    assert not app.has_youth_summer_voucher
+    assert not app.is_active
+    assert not app.has_activation_link_expired
+
+    RejectedYouthApplicationFactory.create(
+        email=app.email,
+        social_security_number=app.social_security_number,
     )
-    assert not inactive_youth_application.has_youth_summer_voucher
-    old_status = inactive_youth_application.status
 
-    assert not inactive_youth_application.is_active
-    assert not inactive_youth_application.has_activation_link_expired
-    assert inactive_youth_application.language == language
+    response = api_client.get(get_activation_url(app.pk))
 
-    if rejected_application_exists:
-        RejectedYouthApplicationFactory.create(
-            email=inactive_youth_application.email,
-            social_security_number=inactive_youth_application.social_security_number,
+    assert response.status_code == status.HTTP_302_FOUND
+    assert response.url == app.additional_info_page_url(app.pk)
+
+    app.refresh_from_db()
+
+    assert not app.has_youth_summer_voucher
+    assert app.is_active
+    assert app.status == YouthApplicationStatus.ADDITIONAL_INFORMATION_REQUESTED
+    assert app.handler is None
+    assert app.handled_at is None
+
+
+@pytest.mark.django_db
+@override_settings(NEXT_PUBLIC_DISABLE_VTJ=False)
+@pytest.mark.parametrize("application_year", [2022, 2023])
+@pytest.mark.parametrize("applicant_age", [15, 16, 17, 18])
+@pytest.mark.parametrize("is_helsinkian", [False, True])
+@pytest.mark.parametrize("attends_helsinkian_school", [False, True])
+@pytest.mark.parametrize("is_alive", [False, True])
+def test_youth_applications_activate_unexpired_inactive__vtj_enabled(
+    api_client,
+    make_youth_application_activation_link_unexpired,
+    settings,
+    application_year,
+    applicant_age,
+    is_helsinkian,
+    attends_helsinkian_school,
+    is_alive,
+):
+    social_security_number = get_random_social_security_number_for_year(
+        year=application_year - applicant_age
+    )
+    application_date = date(year=application_year, month=5, day=1)
+
+    with freeze_time(application_date):
+        app: YouthApplication = (
+            InactiveNoNeedAdditionalInfoYouthApplicationFactory.build(
+                is_unlisted_school=not attends_helsinkian_school,
+                social_security_number=social_security_number,
+                encrypted_handler_vtj_json=None,
+            )
+        )
+        app.encrypted_original_vtj_json = mock_vtj_person_id_query_found_content(
+            first_name=app.first_name,
+            last_name=app.last_name,
+            social_security_number=app.social_security_number,
+            is_alive=is_alive,
+            is_home_municipality_helsinki=is_helsinkian,
         )
+        app.save()
 
-    response = api_client.get(get_activation_url(inactive_youth_application.pk))
+    app.refresh_from_db()
```

Comment:
This seems like an unnecessary refresh as `app` is just saved on line 767(?)

Reply:
Removed and committed. Thanks for the review!